onebusaway-android/src/main/java/org/onebusaway/android/tripservice/TripService.java

```diff
@@ -111,33 +111,34 @@ public TaskContextImpl(int startId) {
         public void taskComplete() {
             //Log.d(TAG, "Task complete: " + mStartId);
             // If we have notifications, then we can't stop ourselves.
-            if (mNotifications.isEmpty()) {
+            //TODO: Is there actually a reason to not stop ourselves if we are no longer storing notifications?
```

Comment:
Hi Sean,

I am not sure that keeping the object as is would work either, since we no
longer locally store notifications (and its now unsafe to do so). The list
would always be empty in every case.

If we need to keep this service alive, some sort of state flag would seem
safer than an empty list of objects.

The important question to ask is if we need to prevent the service from
being killed, and if so, what specific processes need the service kept
alive.

This would seem to me to be safer not only than what we have now but safer
than what we had before. The moment the notification list was empty the
service was always just as vulnerable to being killed as it is now.


On Tue, May 10, 2016 at 10:21 AM, Sean Barbeau <notifications@github.com>
wrote:

> In
> onebusaway-android/src/main/java/org/onebusaway/android/tripservice/TripService.java
> <https://github.com/OneBusAway/onebusaway-android/pull/479#discussion_r62714004>
> :
>
> > @@ -111,33 +111,34 @@ public TaskContextImpl(int startId) {
> >          public void taskComplete() {
> >              //Log.d(TAG, "Task complete: " + mStartId);
> >              // If we have notifications, then we can't stop ourselves.
> > -            if (mNotifications.isEmpty()) {
> > +            //TODO: Is there actually a reason to not stop ourselves if we are no longer storing notifications?
>
> Seems like the simplest solution without a complete overahul of the code
> (e.g. to use something like an IntentService) may be to keep the
> mNotifications object, but only use it as a means to ensure we don't shut
> down the service prematurely. SetLatestEventInfo() should still be
> eliminated.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/OneBusAway/onebusaway-android/pull/479/files/6352fbd746b5be92d461464ffe018ac52552ef7a#r62714004>
>